test/e2e/specs/site-editor/user-global-styles-revisions.spec.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * WordPress dependencies
+ */
+const {
+	test,
+	expect,
+	Editor,
+} = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	editor: async ( { page }, use ) => {
+		await use( new Editor( { page, hasIframe: true } ) );
+	},
+	userGlobalStylesRevisions: async ( { page, requestUtils }, use ) => {
+		await use( new UserGlobalStylesRevisions( { page, requestUtils } ) );
+	},
+} );
+
+test.describe( 'Global styles revisions', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'emptytheme' ),
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.visitSiteEditor( {
+			canvas: 'edit',
+		} );
+	} );
+
+	test( 'should display revisions UI when there is more than 1 revision', async ( {
+		page,
+		editor,
+		userGlobalStylesRevisions,
+	} ) => {
+		// Navigates to Styles -> Typography -> Text and click on a size.
+		await page.getByRole( 'button', { name: 'Styles' } ).click();
```

Comment:
Tip: We can chain locators to write more readable queries so that we don't necessarily need some of the comments:

```suggestion
		await page
			.getByRole( 'region', { name: 'Editor top bar' } )
			.getByRole( 'button', { name: 'Styles' } )
			.click();
```